src/olympia/versions/views.py

```diff
@@ -143,11 +142,7 @@ def download_latest(request, addon, beta=False, type='xpi', platform=None):
         file = sorted(files, key=lambda f: f.platform == platforms[-1])[-1]
     except IndexError:
         raise http.Http404()
-    args = [file.id, type] if type else [file.id]
-    url = posixpath.join(reverse('downloads.file', args=args), file.filename)
-    if request.GET:
-        url += '?' + request.GET.urlencode()
-    return http.HttpResponseRedirect(url)
+    return download_file(request, file.id, type)
```

Comment:
Maybe have `download_file` accept a `file_` keyword argument, defaulting to `None`, and pass `file` here to avoid the extra database query ? Also pass `type`as a keyword argument while you're at it.